### PR TITLE
Luwei/extend reorder test

### PR DESCRIPTION
**Pull request by luweizhou2016**
### Details:
 - *Extend reorder test*

### Tickets:
 - *77001*

### CODE REVIEW

File: src/tests/unit/cpu/CMakeLists.txt [hunk 1]
```diff
@@ -1,22 +1,23 @@
-# Copyright (C) 2018-2022 Intel Corporation
+# Copyright (C) 2018-2021 Intel Corporation
```
Comment (ilyachur) [inline src/tests/unit/cpu/CMakeLists.txt:1]: Why did you change copyrights?

**Comment by dmitry-gorokhov** [comment]
@maxnick could you please review?

**Comment by dmitry-gorokhov** [comment]
@maxnick just a reminder

File: src/tests/unit/cpu/nodes/reorder_node_test.cpp [hunk 2]
```diff
@@ -82,185 +211,373 @@ class ReorderCustomImplTestBase: public ::testing::Test {
...
+        auto parentMemory = std::make_shared<MKLDNNPlugin::MKLDNNMemory>(cpuEngine);
+        auto childMemory = std::make_shared<MKLDNNPlugin::MKLDNNMemory>(cpuEngine);
+        parentMemory->Create(inputDesc, nullptr);
+        childMemory->Create(outputDesc, nullptr);
+        parentEdge->reuse(parentMemory);
+        childEdge->reuse(childMemory);
+
+        reorderNode->setDescs(inputDesc, outputDesc);
+
+        std::vector<std::shared_ptr<MKLDNNPlugin::MKLDNNNode>> nodes{inputNode, reorderNode, outputNode};
```
Comment (maxnick) [inline src/tests/unit/cpu/nodes/reorder_node_test.cpp:495]: Static array may be used.

File: src/tests/unit/cpu/nodes/reorder_node_test.cpp [hunk 3]
```diff
@@ -82,185 +211,373 @@ class ReorderCustomImplTestBase: public ::testing::Test {
...
+        for (auto& n : nodes) {
+            n->init();
+            n->getSupportedDescriptors();
+            n->initSupportedPrimitiveDescriptors();
         }
+        //Select inputDesc as primitive descriptor for reorder node
+        reorderNode->selectPrimitiveDescriptorByIndex(0);
+        stream = {cpuEngine};
+        return;
```
Comment (maxnick) [inline src/tests/unit/cpu/nodes/reorder_node_test.cpp:504]: M.b. it is a matter of taste, but this `return` statement looks unnecessary.

File: src/tests/unit/cpu/nodes/reorder_node_test.cpp [hunk 4]
```diff
@@ -82,185 +211,373 @@ class ReorderCustomImplTestBase: public ::testing::Test {
...
+
+    MKLDNNPlugin::Shape srcDims;
+    MKLDNNPlugin::Shape dstDims;
+    LayoutType srcLayout;
+    LayoutType dstLayout;
+    InferenceEngine::Precision prec;
+    std::vector<std::vector<size_t>> inputShapes;
+    ngraph::PartialShape inputPartialShape;
+    BlockedDescCreator::CreatorsMap blockCreatorMap;
+    MKLDNNPlugin::CpuBlockedMemoryDesc inputDesc{InferenceEngine::Precision::FP32, MKLDNNPlugin::Shape{}};
```
Comment (maxnick) [inline src/tests/unit/cpu/nodes/reorder_node_test.cpp:523]: It looks like we do not need to store the most part of this variables in the test object, they can be just passed as parameters for the `buildReorderNode()` method.

File: src/tests/unit/cpu/nodes/reorder_node_test.cpp [hunk 5]
```diff
@@ -82,185 +211,373 @@ class ReorderCustomImplTestBase: public ::testing::Test {
...
-const std::vector<bool> forceInplace {false, true};
-const auto NSPC2NCSPparams =::testing::Combine(
-                ::testing::Values(std::vector<size_t> {2, 16, 8, 8}),
-                ::testing::ValuesIn(forceInplace));
+const auto reorderCpuTestParams_1 =
+    ::testing::Values(ReorderCPUTestParamSetF32{{2, 16, 8, -1},
+                                                {{2, 16, 8, 8}, {2, 16, 8, 16}, {2, 16, 8, 8}},
+                                                LayoutType::nspc,
+                                                LayoutType::ncsp,
+                                                InferenceEngine::Precision::FP32});
```
Comment (maxnick) [inline src/tests/unit/cpu/nodes/reorder_node_test.cpp:542]: Can we have fully undefined shape {-1, -1, -1, -1} with target shapes with dimensions changing on all axis?

**Comment by luweizhou2016** [comment]
@maxnick  Applied review comments!

**Comment by maxnick** [comment]
@luweizhou2016 Could you please apply or respond to the rest of the review comments?

**Comment by luweizhou2016** [comment]
@maxnick , Just found that I missed some hidden review comments. Will update once applied.

**Comment by luweizhou2016** [comment]
@maxnick ,Applied and  updated with removing the template code for easiness. Thx!

**Comment by luweizhou2016** [comment]
@maxnick , Updated.

File: src/tests/unit/cpu/nodes/reorder_node_test.cpp [hunk 6]
```diff
@@ -1,266 +1,430 @@
...
-#include "nodes/input.h"
-#include <edge.h>
+#include <memory_desc/cpu_memory_desc_utils.h>
+#include <memory_desc/dnnl_memory_desc.h>
 #include <node.h>
+#include <nodes/reorder.h>
+
+#include <common/memory_desc_wrapper.hpp>
+#include <dnnl.hpp>
+#include <typeinfo>
```
Comment (maxnick) [inline src/tests/unit/cpu/nodes/reorder_node_test.cpp:16]: I'm not sure this include is still needed. Could you please check the other includes on whether they are still necessary?

File: src/tests/unit/cpu/nodes/reorder_node_test.cpp [hunk 7]
```diff
@@ -1,266 +1,430 @@
...
+        const auto& reorderInputDesc = inputDesc;
+
+        const ov::intel_cpu::CpuBlockedMemoryDesc outputDesc =
+            dstBlockedDescCreator->createDesc(prec, reorderParams.dstShape);
+
+        buildReorderGraph(reorderInputDesc, outputDesc, reorderParams.srcShape, reorderParams.dstShape);
+    }
+
+private:
+    ov::intel_cpu::CpuBlockedMemoryDesc inputDesc{InferenceEngine::Precision::FP32, ov::intel_cpu::Shape{}};
```
[the remark this answers is not in the document]
Comment (luweizhou2016) [reply src/tests/unit/cpu/nodes/reorder_node_test.cpp]: The store is caused by input node default supported memory descriptor is ncsp, which would cause inputNode->redefineOutputMemory fails because of compatibility.  I have just found cpu plugin inputNode can constructed with external memory descriptor argument, which makes this reorder test graph has compatible selected memory descriptor.
Already updated.

**Comment by luweizhou2016** [comment]
@maxnick , Applied, Thx!

**Review by usstq — APPROVED** [review]
